pallets/gear/src/manager/journal.rs

```diff
@@ -619,4 +619,46 @@ where
     ) {
         ExtManager::send_signal(self, message_id, destination, err)
     }
+
+    fn pay_rent(&mut self, payer: ProgramId, program_id: ProgramId, block_count: u32) {
+        // Querying actual block author to reward.
+        let block_author = Authorship::<T>::author()
+            .unwrap_or_else(|| unreachable!("Failed to find block author!"));
+
+        let from = <T::AccountId as Origin>::from_origin(payer.into_origin());
+        let block_count = block_count.unique_saturated_into();
+        let value = Pallet::<T>::rent_fee_for(block_count);
+
+        ProgramStorageOf::<T>::update_active_program(program_id, |program| {
+            CurrencyOf::<T>::transfer(
+                &from,
+                &block_author,
+                value,
+                ExistenceRequirement::AllowDeath,
+            )
+            .unwrap_or_else(|e| unreachable!("Failed to transfer value: {:?}", e));
```

Comment:
I'd avoid panics here and made `pay_rent` fallible

Reply:
Agree that panics should be avoided. However value cannot be transferred directly within syscall but used to reduce ValueCounter which is set to the actual free balance of the program before execution. If ValueCaounter cannot be updated the `pay_rent` fails with an error.

So this call in JournalNote processing should not fail ever.

Reply:
Yep, panicking here is totally pre-checked, but check my previous comments related to different kind of overflows and saturating operations.
E.g. program payed till u32::MAX / 2 and in case of payment for u32::MAX more we will lose u32::MAX - u32::MAX/2 funds, cause it already transferred above, but bn didn't increased properly.